app/helpers/comments_helper.rb

```diff
@@ -111,11 +110,21 @@ def show_hide_comments_link(commentable, options={})
   #### HELPERS FOR CHECKING WHICH BUTTONS/FORMS TO DISPLAY #####
 
   def can_reply_to_comment?(comment)
```

Comment:
Do you also want to add checks to the `InboxController#reply` action, in case someone is blocked after they load their inbox but before they click the "Reply" button?

Reply:
Sure, if it's not super fussy!

Reply:
Doing a redirect in that particular case isn't too fussy, I think.

It would be a bit fussier to get it to display an error when javascript is enabled, instead of just failing silently. But since that's the existing behavior of the "Reply" button in comment views, I figured it'd be okay not to handle that case.

(It's a little odd that we're only performing this one check, when there are a bunch of others from the `CommentsController` that should also be performed, like disabled comments and frozen comments and replying to spam. But I think the best way to address that is to move all of those checks into the `CommentPolicy` class and rely on Pundit for both users and admins, which is not really in scope.)